operator/pkg/awsprovider/iam/reconciler.go

```diff
@@ -0,0 +1,210 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package iam
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/iam"
+	apis "github.com/awslabs/kit/operator/pkg/apis/controlplane/v1alpha1"
+	"github.com/awslabs/kit/operator/pkg/awsprovider"
+	"github.com/awslabs/kit/operator/pkg/errors"
+	"github.com/awslabs/kit/operator/pkg/kubeprovider"
+	"go.uber.org/zap"
+
+	"knative.dev/pkg/ptr"
+)
+
+type Controller struct {
+	iam        *awsprovider.IAM
+	kubeClient *kubeprovider.Client
+}
+
+// NewController returns a controller for managing LaunchTemplates and ASG in AWS
+func NewController(iam *awsprovider.IAM, client *kubeprovider.Client) *Controller {
+	return &Controller{iam: iam, kubeClient: client}
+}
+
+var (
+	desiredPolicies = []string{
```

Comment:
variable name sounds too generic here. In reality these are policies required for `kitnoderole`